scripts/buildChainsJson.js

```diff
@@ -0,0 +1,104 @@
+const path = require('path');
+const { writeFile } = require('fs/promises');
+const axios = require('axios');
+
+const NOVA_CONFIG_VERSION = process.env.NOVA_CONFIG_VERSION;
+const TELENOVA_CONFIG_VERSION = process.env.TELENOVA_CONFIG_VERSION;
+const CONFIG_PATH = `chains/${TELENOVA_CONFIG_VERSION}/`;
+const TELENOVA_CONFIG_URL = `https://raw.githubusercontent.com/novasamatech/nova-utils/master/chains/${NOVA_CONFIG_VERSION}/`;
+
+const CHAINS_ENV = ['chains_dev.json', 'chains.json'];
+const ALLOWED_CHAINS = require('./data/allowedChains.json');
+
+async function getDataViaHttp(url, filePath) {
+  try {
+    const response = await axios.get(url + filePath);
+
+    return response.data;
+  } catch (error) {
+    console.log('Error: ', error?.message || 'getDataViaHttp failed');
+  }
+}
+
+function fillAssetData(chain) {
+  const assetsList = [];
+  const allowedAssets = ALLOWED_CHAINS[chain.chainId]?.assets || [];
+
+  chain.assets.map(asset => {
+    const allowedAsset = allowedAssets.find(a => a.symbol === asset.symbol);
+    if (allowedAsset) {
+      assetsList.push({
+        assetId: asset.assetId,
+        symbol: asset.symbol,
+        precision: asset.precision,
+        type: asset.type,
+        priceId: asset.priceId,
+        name: allowedAsset.name,
+      });
+    }
+  });
+  return assetsList;
+}
+
+
+function getTransformedData(rawData) {
+  const filteredData = rawData.filter(chain => {
+    const isAllowedChain = chain.chainId in ALLOWED_CHAINS;
+    return isAllowedChain;
+  });
+
+  return filteredData.map((chain, index) => {
+    const assets = fillAssetData(chain)
+    const nodes = chain.nodes
+      .filter(node => !node.url.includes('{'))
+      .map(node => ({
+        url: node.url,
+        name: node.name
+      }));
+
+    const updatedChain = {
+      name: chain.name,
+      chainIndex: index,
```

Comment:
chainIndex was added as an increment, the order will be the same as the order in the chains.json file. This index can change for a chain, if we put a new chain in the middle of the list, the tail of the list will be re-indexed. The order will be taken from nova-utils.

Reply:
```javascript
function getTransformedData(rawData) {
  const filteredData = rawData.filter(chain => chain.chainId in ALLOWED_CHAINS);

  return filteredData.map((chain, index) => {
    const assets = fillAssetData(chain);
    const nodes = chain.nodes
      .filter(node => !node.url.includes('{'))
      .map(node => ({ url: node.url, name: node.name }));

    return {
      name: chain.name,
      chainIndex: index,
      addressPrefix: chain.addressPrefix,
      chainId: `0x${chain.chainId}`,
      assets,
      nodes,
    };
  });
}

```